src/network/local_socket_reply.cpp

```diff
@@ -0,0 +1,232 @@
+    http_data += op;
+    http_data += ' ';
+    http_data += request.url().path();
+    http_data += " HTTP/1.1\r\n";
+
+    // Build the HTTP Host header
+    // Host can be anything, so we'll use 'multipass'
+    http_data += "Host: multipass\r\n";
+
+    // Build the HTTP User-Agent header
+    // We'll just use what Qt uses
+    http_data += "User-Agent: Mozilla/5.0\r\n";
+
+    // Build the HTTP Accept header
+    // Default to accept everything
+    http_data += "Accept: */*\r\n";
+
+    if (op == "POST" || op == "PUT")
+    {
+        http_data += "Content-Type: application/x-www-form-urlencoded\r\n";
+
+        if (outgoingData)
+        {
+            outgoingData->open(QIODevice::ReadOnly);
+
+            http_data += "Content-Length: ";
+            http_data += QByteArray::number(outgoingData->size());
+            http_data += "\r\n\r\n";
+            http_data += outgoingData->readAll();
+        }
+    }
+
+    http_data += "\r\n";
+
+    local_socket->write(http_data);
+    local_socket->flush();
+}
+
+void mp::LocalSocketReply::read_reply()
+{
+    local_socket->read(reply_data.data(), len);
+
+    parse_reply();
+
+    emit finished();
+}
+
+void mp::LocalSocketReply::parse_reply()
+{
+    auto reply_lines = reply_data.split('\n');
+    auto it = reply_lines.constBegin();
+
+    parse_status(*it);
+
+    for (++it; it != reply_lines.constEnd(); ++it)
+    {
+        if ((*it).contains("Transfer-Encoding") && (*it).contains("chunked"))
+            chunked_transfer_encoding = true;
+
+        if ((*it).isEmpty() || (*it).startsWith('\r'))
+        {
+            // Advance to the body
+            // Chunked transfer encoding also includes a line with the amount of
+            // bytes (in hex) in the chunk. We just skip it for now.
+            it = chunked_transfer_encoding ? it + 2 : it + 1;
+
+            content_data = (*it).trimmed();
+
+            break;
+        }
+    }
+}
+
+// Some of this logic is taken from Qt src/network/access/qhttpnetworkreply.cpp
+void mp::LocalSocketReply::parse_status(const QByteArray& status)
+{
+    const int minLength = 11;
+    const int dotPos = 6;
+    const int spacePos = 8;
+    const char httpMagic[] = "HTTP/";
+
+    if (status.length() < minLength || !status.startsWith(httpMagic) || status.at(dotPos) != '.' ||
+        status.at(spacePos) != ' ')
+    {
+        setError(QNetworkReply::ProtocolFailure, "Malformed HTTP response from server");
+
+        emit error(QNetworkReply::ProtocolFailure);
+
+        return;
+    }
+
+    int i = spacePos;
+    int j = status.indexOf(' ', i + 1); // j == -1 || at(j) == ' ' so j+1 == 0 && j+1 <= length()
+    const QByteArray code = status.mid(i + 1, j - i - 1);
+
+    bool ok;
+    auto statusCode = code.toInt(&ok);
+
+    if (statusCode >= 400)
+    {
+        auto error_code = statusCodeFromHttp(statusCode);
+
+        setError(error_code, QString::fromLatin1(status.constData() + j + 1));
+
+        emit error(error_code);
+    }
+}
```

Comment:
Couldn't we just regex this?…

Reply:
Sure, many ways to do this.  I thought the fastest way was to just copy what Qt already does in its implementation :shrug:  Didn't feel the need to reinvent the wheel.

Reply:
It's just that someone needs to review it and maintain ;)

Reply:
@Saviq, hey, your regex foo is much better than mine, so I'm asking for some help on this if you'd rather this be a regex :grin:  Otherwise, this works and I'd rather spend my time getting this PR good to go and getting #1533 done.

Reply:
Oh sure, https://regex101.com/r/FJVUf8/1 :)

Reply:
Or well, https://regex101.com/r/FJVUf8/2/ to be exactly as the above, but I don't think the status code should be anything else than 3 digits?

Reply:
Man, your good:)  Yes, afaik, http status codes should be 3 digits, so I think it's safe to delineate on that or otherwise it's a malformed status code.  Thanks!

Reply:
Done